jujugui/static/gui/src/app/components/deployment-flow/cloud/cloud.js

```diff
@@ -25,34 +25,64 @@ YUI.add('deployment-cloud', function() {
       acl: React.PropTypes.object.isRequired,
       cloud: React.PropTypes.string,
       clouds: React.PropTypes.object.isRequired,
+      listClouds: React.PropTypes.func.isRequired,
       setCloud: React.PropTypes.func.isRequired
     },
 
+    getInitialState: function() {
+      return {
+        clouds: [],
+        cloudsLoading: true
+      };
+    },
+
+    componentWillMount: function() {
+      this.props.listClouds((error, clouds) => {
+        if (error) {
+          console.error('Unable to list clouds', error);
+          return;
+        }
+        if (!clouds) {
```

Comment:
Ahh ok, maybe a code comment or explicit check is necessary then, for future us who go back and put in the default value :)